src/components/MapWithPoints.vue

```diff
@@ -0,0 +1,179 @@
+} from "vue2-leaflet";
+
+interface Point {
+  name: string;
+  location: LatLng;
+}
+
+export default {
+  name: "MapWithPoints",
+  components: {
+    LMap,
+    LTooltip,
+    LCircle,
+    LCircleMarker,
+    LWMSTileLayer,
+    LControlLayers,
+  },
+  props: {
+    points: {
+      type: Array,
+      required: true,
+    },
+    radius: {
+      type: Number,
+      default: 0,
+    },
+    height: {
+      type: Number,
+      default: 400,
+    },
+    zoom: {
+      type: Boolean,
+      default: true,
+    },
+    canChangeBaseMap: {
+      type: Boolean,
+      default: true,
+    },
+    isInteractive: {
+      type: Boolean,
+      default: true,
+    },
+    navigateToPoint: {
+      type: Function,
+      required: false,
+    },
+  },
+  computed: {
+    mapLayers() {
+      const OpenStreetMapFallbackLayer = {
+        name: "OpenStreetMap Basemap",
+        visible: false,
+        url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+        attribution:
+          '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+      };
+      if (linzBasemapApiKey && linzBasemapApiKey !== "YOUR_API_KEY_HERE") {
+        return [
+          {
+            name: "LINZ Basemap",
+            visible: true, // Make the LINZ basemap the default one
+            attribution:
+              '<a href="//www.linz.govt.nz/data/linz-data/linz-basemaps/data-attribution">LINZ CC BY 4.0 © Imagery Basemap contributors</a>',
+            url: `https://basemaps.linz.govt.nz/v1/tiles/aerial/3857/{z}/{x}/{y}.webp?api=${linzBasemapApiKey}`,
+          },
+          {
+            name: "LINZ Topo",
+            visible: false, // Make the LINZ basemap the default one
+            attribution:
+              '<a href="//www.linz.govt.nz/data/linz-data/linz-basemaps/data-attribution">LINZ CC BY 4.0 © Imagery Basemap contributors</a>',
+            url: `http://tiles-a.data-cdn.linz.govt.nz/services;key=${linzBasemapApiKey}/tiles/v4/layer=767/EPSG:3857/{z}/{x}/{y}.png`,
+          },
+          OpenStreetMapFallbackLayer,
+        ];
+      }
+      return [{ ...OpenStreetMapFallbackLayer, visible: true }];
+    },
+    mapBounds() {
+      // Calculate the initial map bounds and zoom level from the set of lat/lng points
+      return latLngBounds([
+        this.mapLocations.map(({ location }) => location),
+      ]).pad(0.25);
+    },
+    mapLocations(): Point[] {
+      return this.points.map(({ location, ...rest }) => ({
+        location:
+          typeof location === "string"
+            ? latLng(
+                location.split(", ").map(Number) as [number, number, number]
+              )
+            : location,
+        ...rest,
+      }));
+    },
+    mapLocationsForRadius(): Point[] {
+      if (this.radius !== 0) {
+        return this.mapLocations;
+      }
+      return [];
+    },
+    mapHeight(): string {
+      return `${this.height}px`;
+    },
+  },
+  methods: {
+    navigateToLocation(point: Point) {
```

Comment:
would be nice to support new tab opening, i think this will do the trick
`      if (!(event.metaKey || event.ctrlKey || event.shiftKey)) {`